lib/rubocop/cop/sorbet/signatures/runtime_on_failure_depends_on_checked.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Sorbet
+      # Checks that `on_failure` is not used without `checked(:tests)` or `checked(:always)`.
+      class RuntimeOnFailureDependsOnChecked < ::RuboCop::Cop::Base
+        include SignatureHelp
+
+        MSG = "To use .on_failure you must additionally call .checked(:tests) or .checked(:always), otherwise, the .on_failure has no effect."
+
+        def_node_matcher :on_failure_call?, <<~PATTERN
+          (send _ :on_failure ...)
+        PATTERN
+
+        def_node_matcher :checked_tests_or_always?, <<~PATTERN
+          (send _ :checked (sym {:tests | :always}))
+        PATTERN
+
+        def on_signature(node)
+          return unless node.descendants.any? { |n| on_failure_call?(n) }
+          return if node.descendants.any? { |n| checked_tests_or_always?(n) }
+
+          add_offense(node, message: MSG)
```

Comment:
Using `MSG` is already the default behaviour:

```suggestion
          add_offense(node)
```